#VRML_SIM R2022a utf8
WorldInfo {
  info [
    "Description"
    "Author: first name last name <e-mail>"
    "Date: DD MMM YYYY"
  ]
}
Viewpoint {
  orientation 0.3293903142052062 0.7565678181808763 -0.5648957048879238 1.3132568108891618
  position -5.211309614006818 5.64253660138002 11.485759300951381
  follow "epuck0"
  followType "None"
}
PointLight {
  attenuation 0 0 1
  intensity 0.5
  location 0.55 1.06 0.58
  castShadows TRUE
}
PointLight {
  attenuation 0 0 1
  intensity 0.5
  location 0.05 1.63 0.53
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
DEF ground Solid {
  translation 0 1.5 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          ambientIntensity 1
          diffuseColor 1 1 1
          emissiveColor 0.823529 0.6 0.34902
          shininess 0
          specularColor 0.196078 0.196078 0
        }
        texture ImageTexture {
          url [
            "textures/lightwood.png"
          ]
        }
        textureTransform TextureTransform {
          center 0.46 0
          rotation -0.32
          translation -1.01 0
        }
      }
      geometry DEF ground_geo IndexedFaceSet {
        coord Coordinate {
          point [
            2 1.6 0
            -2 1.6 0
            -2 -4.6 0
            2 -4.6 0
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0 5
            5 5
            5 0
          ]
        }
        coordIndex [
          0, 1, 2, 3, -1
        ]
        texCoordIndex [
          0, 1, 2, 3, -1
        ]
      }
    }
  ]
  boundingObject USE ground_geo
}
DEF long_rock_3 Solid {
  translation 0 3 0
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.01 0.7 0.2
      }
    }
  ]
  name "short brick"
  boundingObject DEF U_END_BOX Box {
    size 0.01 0.6 0.2
  }
}
DEF long_rock_up Solid {
  translation -2.93856e-07 3.08 0
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.01 4 0.2
      }
    }
  ]
  name "short brick(1)"
  boundingObject DEF U_END_BOX Box {
    size 0.01 4 0.2
  }
}
DEF long_rock_left Solid {
  translation 1.98001 -0.0199894 0
  rotation 0 0 -1 -5.307179586466759e-06
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.01 6.3 0.2
      }
    }
  ]
  name "short brick(3)"
  boundingObject DEF U_END_BOX Box {
    size 0.01 6.3 0.2
  }
}
DEF long_rock_left Solid {
  translation -1.97999 -0.0200109 0
  rotation 0 0 -1 -5.307179586466759e-06
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.01 6.3 0.2
      }
    }
  ]
  name "short brick(4)"
  boundingObject DEF U_END_BOX Box {
    size 0.01 6.3 0.2
  }
}
DEF long_rock_bottom Solid {
  translation 2.2333e-05 -3.08 0
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.01 4 0.2
      }
    }
  ]
  name "short brick(2)"
  boundingObject DEF U_END_BOX Box {
    size 0.01 4 0.2
  }
}
DEF rock Solid {
  translation 0.22167 0.267548 0
  rotation 0 0 1 5.02662
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick1"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation 1.06532 1.38774 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick2"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation -0.47529 -1.25364 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick3"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation -0.16115 1.53541 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick4"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation -0.776227 1.34035 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick5"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation 0.0741107 2.34293 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick6"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation 1.32743 -1.09992 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.05 0.05 0.2
      }
    }
  ]
  name "short brick7"
  boundingObject DEF U_END_BOX Box {
    size 0.05 0.05 0.2
  }
}
DEF rock Solid {
  translation 1.3463 0.105938 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short bric8"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation -1.31682 -0.931062 0
  rotation 0 0 -1 0.325642
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick9"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation -0.846635 0.0855468 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick10"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation -0.107609 -1.91494 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick11"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation 0.127712 -0.958022 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick12"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation -1.72366 1.05597 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick13"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF rock Solid {
  translation 0.686015 1.95487 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.37 0.15 0.2
      }
    }
  ]
  name "short brick14"
  boundingObject DEF U_END_BOX Box {
    size 0.37 0.15 0.2
  }
}
DEF rock Solid {
  translation -0.993985 1.95487 0
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.37 0.15 0.2
      }
    }
  ]
  name "short brick14(1)"
  boundingObject DEF U_END_BOX Box {
    size 0.37 0.15 0.2
  }
}
DEF rock Solid {
  translation 0.478731 1.20749 0
  rotation 0 0 -1 2.05704
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 1
        }
        texture ImageTexture {
          url [
            "textures/stone-2.png"
          ]
        }
      }
      geometry Box {
        size 0.15 0.15 0.2
      }
    }
  ]
  name "short brick15"
  boundingObject DEF U_END_BOX Box {
    size 0.15 0.15 0.2
  }
}
DEF migration_target Solid {
  translation 0 0 -1
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.45098 0.823529 0.0862745
        transparency 0.3
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.001
        radius 0.3
      }
    }
  ]
  name "migration_target"
}
DEF epuck0 E-puck {
  translation 0 2.9 0
  rotation 0 0 1 -1.5707996938995747
  name "epuck0"
  controller "collective_pso"
  emitter_channel 0
  receiver_channel 0
  groundSensorsSlot [
    GPS {
      translation -0.03 0 0
    }
  ]
}
DEF epuck1 E-puck {
  translation 0.13 2.9 0
  rotation 0 0 1 -1.5707996938995747
  name "epuck1"
  controller "collective_pso"
  emitter_channel 0
  receiver_channel 0
  groundSensorsSlot [
    GPS {
      translation -0.03 0 0
    }
  ]
}
DEF epuck2 E-puck {
  translation -0.13 2.9 0
  rotation 0 0 1 -1.5707996938995747
  name "epuck2"
  controller "collective_pso"
  emitter_channel 0
  receiver_channel 0
  groundSensorsSlot [
    GPS {
      translation -0.03 0 0
    }
  ]
}
DEF epuck3 E-puck {
  translation 0.26 2.9 0
  rotation 0 0 1 -1.5707996938995747
  name "epuck3"
  controller "collective_pso"
  emitter_channel 0
  receiver_channel 0
  groundSensorsSlot [
    GPS {
      translation -0.03 0 0
    }
  ]
}
DEF epuck4 E-puck {
  translation -0.27 2.9 0
  rotation 0 0 1 -1.5707996938995747
  name "epuck4"
  controller "collective_pso"
  emitter_channel 0
  receiver_channel 0
  groundSensorsSlot [
    GPS {
      translation -0.03 0 0
    }
  ]
}
Robot {
  children [
    Emitter {
      name "emitter_loc"
    }
  ]
  name "supervisor"
  controller "flocking_super_pso"
  controllerArgs [
    ""
  ]
  supervisor TRUE
}
